crates/sui-indexer/src/store/pg_indexer_store.rs

```diff
@@ -1337,12 +1337,13 @@ impl IndexerStore for PgIndexerStore {
         object_deletion_latency: Histogram,
     ) -> Result<(), IndexerError> {
         transactional_blocking!(&self.blocking_cp, |conn| {
-            // update epoch transaction count
+            // update epoch transaction count within an epoch;
+            // do no update after end-of-epoch commit to avoid double counting
             let sql = "UPDATE epochs e1
-SET epoch_total_transactions = e2.epoch_total_transactions + $1
-FROM epochs e2
-WHERE e1.epoch = e2.epoch
-  AND e1.epoch = $2;";
+                        SET epoch_total_transactions = e2.epoch_total_transactions + $1
```

Comment:
shouldn't we remove this because now we only calculate epoch tx count at the end of the epoch?

Reply:
we keep this to reveal total tx count before the end of the epoch, it's actually cleaner if we can remove this one, b/c today there are often write racing conditions and takes retries to commit.